$OpenBSD: patch-src_transform_c,v 1.1 2014/01/05 10:43:49 jasper Exp $

Security fix for CVE-2013-6412,
augeas: incorrect permissions set on newly created files

--- src/transform.c.orig	Sun Jan  5 11:22:51 2014
+++ src/transform.c	Sun Jan  5 11:23:07 2014
@@ -1114,7 +1114,7 @@ int transform_save(struct augeas *aug, struct tree *xf
         mode_t curumsk = umask(022);
         umask(curumsk);
 
-        if (fchmod(fileno(fp), 0666 - curumsk) < 0) {
+        if (fchmod(fileno(fp), 0666 & ~curumsk) < 0) {
             err_status = "create_chmod";
             return -1;
         }
